src/map.cpp

```diff
@@ -6726,13 +6727,38 @@ void map::remove_rotten_items( Container &items, const tripoint &pnt )
     const tripoint abs_pnt = getabs( pnt );
     for( auto it = items.begin(); it != items.end(); ) {
         if( has_rotten_away( *it, abs_pnt ) ) {
+            //If the item that is rotting is a food item, see if we can spawn a monster
+            if (it->is_comestible()){
+                rotten_item_spawn(it, pnt);
+            }
+            //Remove the item regardless
             it = i_rem( pnt, it );
         } else {
             ++it;
         }
     }
 }
 
+template <typename Item>
+void map::rotten_item_spawn( Item &item, const tripoint &pnt )
+{
+        auto &comest = item->type->comestible;
+        std::string mgroup = comest->rot_spawn;
+        if ( mgroup != "null" ) {
+                int chance = comest->rot_spawn_chance;
+                chance *= get_option<int>( "CARRION_SPAWNRATE" )/100;
```

Comment:
Those two lines can be combined into one. In fact, they should be because currently any value of "CARRION_SPAWNRATE" below 100 will set the chance to 0: `get_option<int>` returns an `int` and `100` is an `int` as well, so the division is done as integer division (e.g. 80/100 == 0) and the multiplication is only done after that.

I suggest this: `const int chance = ( rot_spawn_chance * get_option ) / 100;`